packages/geo/src/lib/layer/layer-item/layer-item.component.ts

```diff
@@ -28,6 +28,8 @@ export class LayerItemComponent implements OnInit, OnDestroy {
 
   @Input() toggleLegendOnVisibilityChange: boolean = false;
 
+  @Input() expandLegendVisibleLayers: boolean = false;
```

Comment:
In the context of a layer item, this input should be name differently because there is no concept of "layers" (plural). I would name it: expandLegendIfVisible.